ynr/apps/sopn_parsing/tests/test_sopn_helpers.py

```diff
@@ -88,8 +89,6 @@ def test_multipage_doc(self):
         doc = SOPNDocument(open(example_doc_path, "rb"))
 
         self.assertEqual(len(doc.pages), 9)
-
-        na_wards = doc.get_pages_by_ward_name("north antrim")
-
-        self.assertEqual(len(na_wards), 5)
-        self.assertEqual(na_wards[0].page_number, 5)
+        na_wards = doc.get_pages_by_ward_name("ulster")
+        self.assertEqual(len(na_wards), 9)
+        self.assertEqual(na_wards[0].page_number, 1)
```

Comment:
The previous version of this test was failing legitimately: The correct pages had no been matched as per the PDF. 

That is, in our test PDF the Ulster wards span pages 1 to 4 and the North Antrim ones span pages 5 to 9.

The tests started failing because of the logic in `detect_top_page`, but also we can see the bug in this new test: it's stating that the wards relating to "ulster" span pages 1 to 9 in the PDF, but that's all the pages and we know there to be two wards in the PDF.

This is because the first page of every PDF is always identical to the top of the document, so it was getting marked as a continuation page. The nice thing is we can special case this, because we know the first page is never a continuation page. I've fixed this by adding the following:

```diff
diff --git ynr/apps/sopn_parsing/helpers/pdf_helpers.py ynr/apps/sopn_parsing/helpers/pdf_helpers.py
index 0218d0ed..4599fb40 100644
--- ynr/apps/sopn_parsing/helpers/pdf_helpers.py
+++ ynr/apps/sopn_parsing/helpers/pdf_helpers.py
@@ -123,10 +123,16 @@ class SOPNPageText:
         assume this is a top page and return True.
 
         """
+        if self.page_number == 1:
+            self.is_top_page = True
+            return self.is_top_page
```


We can also update this test to make it a little better and make sure we're matching the correct pages:

```diff
diff --git ynr/apps/sopn_parsing/tests/test_sopn_helpers.py ynr/apps/sopn_parsing/tests/test_sopn_helpers.py
index af809cfe..4c7faa26 100644
--- ynr/apps/sopn_parsing/tests/test_sopn_helpers.py
+++ ynr/apps/sopn_parsing/tests/test_sopn_helpers.py
@@ -87,8 +87,12 @@ class TestSOPNHelpers(TestCase):
             join(dirname(__file__), "data/NI-Assembly-Election-2016.pdf")
         )
         doc = SOPNDocument(open(example_doc_path, "rb"))
-
         self.assertEqual(len(doc.pages), 9)
-        na_wards = doc.get_pages_by_ward_name("ulster")
-        self.assertEqual(len(na_wards), 9)
-        self.assertEqual(na_wards[0].page_number, 1)
+        na_wards = doc.get_pages_by_ward_name("north antrim")
+        self.assertEqual(len(na_wards), 5)
+        self.assertEqual(na_wards[0].page_number, 5)
+
+        doc = SOPNDocument(open(example_doc_path, "rb"))
+        ulster_wards = doc.get_pages_by_ward_name("ulster")
+        self.assertEqual(len(ulster_wards), 4)
+        self.assertEqual(ulster_wards[0].page_number, 1)
```